tooling/debugger/src/lib.rs

```diff
@@ -74,25 +79,77 @@ impl<'backend, B: BlackBoxFunctionSolver> DebugContext<'backend, B> {
         } else {
             println!("Stopped at opcode {}: {}", ip, opcodes[ip]);
-            Self::show_source_code_location(&OpcodeLocation::Acir(ip), &self.debug_artifact);
+            self.show_source_code_location(&OpcodeLocation::Acir(ip), &self.debug_artifact);
         }
     }
 
-    fn show_source_code_location(location: &OpcodeLocation, debug_artifact: &DebugArtifact) {
+    fn print_location_path(&self, loc: Location) {
+        let line_number = self.debug_artifact.location_line_number(loc).unwrap();
+        let column_number = self.debug_artifact.location_column_number(loc).unwrap();
+
+        println!(
+            "At {}:{line_number}:{column_number}",
+            Files::name(&self.debug_artifact, loc.file).unwrap()
+        );
+    }
+
+    fn show_source_code_location(&self, location: &OpcodeLocation, debug_artifact: &DebugArtifact) {
         let locations = debug_artifact.debug_symbols[0].opcode_location(location);
         if let Some(locations) = locations {
             for loc in locations {
-                let file = &debug_artifact.file_map[&loc.file];
-                let source = &file.source.as_str();
-                let start = loc.span.start() as usize;
-                let end = loc.span.end() as usize;
-                println!("At {}:{start}-{end}", file.path.as_path().display());
-                println!(
-                    "\n{}{}{}\n",
-                    &source[0..start].to_string().dimmed(),
-                    &source[start..end],
-                    &source[end..].to_string().dimmed(),
-                );
+                self.print_location_path(loc);
+
+                let loc_line_index = debug_artifact.location_line_index(loc).unwrap();
+
+                // How many lines before or after the location's line we
+                // print
+                let context_lines = 5;
+
+                let first_line_to_print =
+                    if loc_line_index < context_lines { 0 } else { loc_line_index - context_lines };
+
+                let last_line_index = debug_artifact.last_line_index(loc).unwrap();
+                let last_line_to_print = if loc_line_index + context_lines > last_line_index {
+                    last_line_index
+                } else {
+                    loc_line_index + context_lines
+                };
+
+                let source = debug_artifact.location_source_code(loc).unwrap();
+                for (current_line_index, line) in source.lines().enumerate() {
+                    let current_line_number = current_line_index + 1;
+
+                    if current_line_index < first_line_to_print {
+                        // Ignore lines before range starts
+                        continue;
+                    } else if current_line_index == first_line_to_print && current_line_index > 0 {
+                        // Denote that there's more lines before but we're not showing them
+                        print_ellipsized_line(current_line_index);
```

Comment:
```suggestion
                        print_line_of_ellipses(current_line_index);
```

"ellipsized" feels a bit weird to me.

Reply:
It's certainly an abuse of language 😅 . Will change it.